src/screens/WalletConnect/CallRequest/TransactionRequestContent.js

```diff
@@ -158,18 +166,31 @@ const useTransactionFee = (request: WalletConnectCallRequest) => {
 
 const useViewData = (request: WalletConnectCallRequest) => {
   const { t } = useTranslation();
+  const activeAccount = useActiveAccount();
+  const isArchanovaAccountDeployed = useRootSelector(isArchanovaAccountDeployedSelector);
+
+  const isArchanovaAccountActive = isArchanovaAccount(activeAccount);
 
   let errorMessage = null;
   const chain = chainFromChainId[request.chainId];
   if (!chain) {
     errorMessage = t('error.walletConnect.cannotDetermineEthereumChain');
   }
 
-  const isDeployedOnChain = useRootSelector(isDeployedOnChainSelector);
+  if (isArchanovaAccountActive && chain === CHAIN.ETHEREUM) {
```

Comment:
Important seems like second condition need to be negated!

```suggestion
  if (isArchanovaAccountActive && chain !== CHAIN.ETHEREUM) {
```

Better yet would use `useSupportedChains` instead of manually figuring out archanova status.